### DISCUSSION TITLE
Default environment variables available during build?

### DISCUSSION THREAD

**Discussion by natevick** [community/community#33731] [Codespaces] [answered]
It does not seem that the default codespaces environment variables are available during the container build. I'm attempting to use $CODESPACES in a conditional for an initializeCommand bash script. I have debugging echoes, and it is bypassing to the non-codespaces of the conditional.

```
if [[ -n "${CODESPACES}" ]]; then
  echo "CODESPACES"
else
  echo "NOT CODESPACES"
fi
```
If they are not available is there another way to determine the build is happening in codespaces?

**Comment by natevick** [comment]
After putting a `printenv` in my script, it seems to confirm the `$CODESPACES` environment variable is not available during build time. I decided to check if `$GITHUB_CODESPACE_TOKEN` is not empty which got me over my hurdle. I would like to hear how others are dealing with checking if a build is in codespaces or not.

**Comment by cdwort** [comment, marked as answer]
Thanks for bringing this to our attention. We do intend for `CODESPACES` as well as `CODESPACES_NAME` to be set on the host VM to allow for logic like this. We consider this a bug and now have an issue for it. I'm sorry you hit this snag and appreciate you writing in!

Of what is currently available to you on the host VM, I agree that `GITHUB_CODESPACE_TOKEN` is your best, most reliable option.

**Reply by natevick** [reply]
Thanks for the reply @cdwort!

**Comment by jkeech** [comment]
Thanks for raising this! We now surface both `$CODESPACES` and `$CODESPACE_NAME` environment variables during the `initializeCommand`. Please let us know if it's not working like you'd expect for new codespaces. Thanks!

**Reply by natevick** [reply]
Thanks @jkeech! I also ran into this using pre-builds. Will the above change help with that scenario as well?

**Reply by jkeech** [reply]
@nvick, `$CODESPACES` is available during prebuilds. `$CODESPACE_NAME` is not present during prebuilds because a prebuild is used as the base for multiple codespaces from that repo and has not been assigned a specific instance name at that point. `$CODESPACE_NAME` is available starting when a user creates a codespace, whether it is from a prebuild or not. Hope that helps!

**Reply by natevick** [reply]
Thank you!
